lib/gs-utils.c

```diff
@@ -1041,4 +1041,20 @@ gs_utils_get_memory_total (void)
 	return si.totalram / MB_IN_BYTES / si.mem_unit;
 }
 
+/**
+ * gs_utils_set_install_timestamp:
+ *
+ * Sets the value of install-timestamp which represents the last time
+ * we had an update or upgrade.
+ *
+ **/
+void
+gs_utils_set_install_timestamp (GSettings *settings)
+{
```

Comment:
Maybe you should also check if the settings are valid here (`g_return_if_fail`). Because you don't check in all places you called the function.